### ISSUE TITLE
Parent Tenant Field Needed on each tenant

### ISSUE THREAD

**Issue by JoshTango** [OrchardCMS/OrchardCore#12565]
I want a tenant to be able to spawn sub tenants.
We need each tenant to have a parent field to keep track of which tenant spawned it.
Here are a few use cases:

An organization is assigned a tenant.  Then that organization wants to create sub tenants called Spaces for each department in its org.
A content site is assigned a tenant.  then the content site wants to create a sub tenant for each channel in its site.
A posting site like reddit is assigned a tenant.  Then that organization wants to create sub tenants for each community.

Since you plan on making Orchard massively scalable, I see no reason why one instance can't have tons of tenants each having tons of subtenants.

**Comment by Skrypt** [comment]
Multiple levels of sub-tenancy support.
This could create potential issues with tenant folders themselves.
If we create a subtenant that has the same name as another tenant then the folder naming convention would need to be rethought by using a UID. The idea is good but it would have a deep impact on the current structure.
Need to see @jtkech thoughts about it.

**Comment by JoshTango** [comment]
I tried to start the idea with #12564 but ran into a different problem.
I was using GUID's for each tenant.

**Comment by Skrypt** [comment]
There is no notion of a third level and up for tenants. Maybe using the Category field to tag your tenants and group them together or if it is possible adding a custom filter of your own in there. But, I think that is all you can do for now. Multiple levels of tenancy is no supported.

**Comment by Skrypt** [comment]
As you can see the Category field of a Tenant is leveraging the AppSettings.json. So, you could add a custom setting in there that would store a UID that would reference a different tenant.

**Comment by JoshTango** [comment]
Yes possibly but currently I need to be able to resolve the ISetup Service #12564

**Comment by JoshTango** [comment]
I don't see what the existing field for NAME cannot be replaced with a PATH.  Then you simply create a hierarchy of folders for subtenants.

**Comment by ns8482e** [comment]
@JoshTango There is such thing as Parent Tenant, All Tenants are parallel and isolated on it's own.  OC can have only one Default tenant.

For some reason, share data between tenants then you use Open ID

**Comment by JoshTango** [comment]
Yes my plan is to eventually use OpenId as well.
